passes/techmap/abc.cc

```diff
@@ -1046,121 +1045,182 @@ void abc_module(RTLIL::Design *design, RTLIL::Module *current_module, std::strin
+					cell->setPort(ID::E, module->wire(name_e));
+					cell->setPort(ID::F, module->wire(name_f));
+					cell->setPort(ID::G, module->wire(name_g));
+					cell->setPort(ID::H, module->wire(name_h));
+					cell->setPort(ID::S, module->wire(name_s));
+					cell->setPort(ID::T, module->wire(name_t));
+					cell->setPort(ID::U, module->wire(name_u));
+					cell->setPort(ID::Y, module->wire(name_y));
 					design->select(module, cell);
 					continue;
 				}
 				if (c->type == ID(MUX16)) {
 					RTLIL::Cell *cell = module->addCell(remap_name(c->name), ID($_MUX16_));
 					if (markgroups) cell->attributes[ID::abcgroup] = map_autoidx;
-					cell->setPort(ID::A, RTLIL::SigSpec(module->wires_[remap_name(c->getPort(ID::A).as_wire()->name)]));
-					cell->setPort(ID::B, RTLIL::SigSpec(module->wires_[remap_name(c->getPort(ID::B).as_wire()->name)]));
-					cell->setPort(ID::C, RTLIL::SigSpec(module->wires_[remap_name(c->getPort(ID::C).as_wire()->name)]));
-					cell->setPort(ID::D, RTLIL::SigSpec(module->wires_[remap_name(c->getPort(ID::D).as_wire()->name)]));
-					cell->setPort(ID::E, RTLIL::SigSpec(module->wires_[remap_name(c->getPort(ID::E).as_wire()->name)]));
-					cell->setPort(ID::F, RTLIL::SigSpec(module->wires_[remap_name(c->getPort(ID::F).as_wire()->name)]));
-					cell->setPort(ID::G, RTLIL::SigSpec(module->wires_[remap_name(c->getPort(ID::G).as_wire()->name)]));
-					cell->setPort(ID::H, RTLIL::SigSpec(module->wires_[remap_name(c->getPort(ID::H).as_wire()->name)]));
-					cell->setPort(ID::I, RTLIL::SigSpec(module->wires_[remap_name(c->getPort(ID::I).as_wire()->name)]));
-					cell->setPort(ID::J, RTLIL::SigSpec(module->wires_[remap_name(c->getPort(ID::J).as_wire()->name)]));
-					cell->setPort(ID::K, RTLIL::SigSpec(module->wires_[remap_name(c->getPort(ID::K).as_wire()->name)]));
-					cell->setPort(ID::L, RTLIL::SigSpec(module->wires_[remap_name(c->getPort(ID::L).as_wire()->name)]));
-					cell->setPort(ID::M, RTLIL::SigSpec(module->wires_[remap_name(c->getPort(ID::M).as_wire()->name)]));
-					cell->setPort(ID::N, RTLIL::SigSpec(module->wires_[remap_name(c->getPort(ID::N).as_wire()->name)]));
-					cell->setPort(ID::O, RTLIL::SigSpec(module->wires_[remap_name(c->getPort(ID::O).as_wire()->name)]));
-					cell->setPort(ID::P, RTLIL::SigSpec(module->wires_[remap_name(c->getPort(ID::P).as_wire()->name)]));
-					cell->setPort(ID::S, RTLIL::SigSpec(module->wires_[remap_name(c->getPort(ID::S).as_wire()->name)]));
-					cell->setPort(ID::T, RTLIL::SigSpec(module->wires_[remap_name(c->getPort(ID::T).as_wire()->name)]));
-					cell->setPort(ID::U, RTLIL::SigSpec(module->wires_[remap_name(c->getPort(ID::U).as_wire()->name)]));
-					cell->setPort(ID::V, RTLIL::SigSpec(module->wires_[remap_name(c->getPort(ID::V).as_wire()->name)]));
-					cell->setPort(ID::Y, RTLIL::SigSpec(module->wires_[remap_name(c->getPort(ID::Y).as_wire()->name)]));
+					RTLIL::IdString name_a = remap_name(c->getPort(ID::A).as_wire()->name);
```

Comment:
Are the intermediate varibles helpful? Especially for large muxes they seem to add clutter, not remove it.

Reply:
It's a good question.  I think they are helpful, because they break up those extremely busy lines and help make clear exactly what is being done.  But you are definitely right that they make the file longer.

My thinking is that short and clear is best, but failing that, long and clear is preferable to short and impenetrable.  I'm open to other opinions, though.

Reply:
I'll leave it to someone else to decide on this.

Reply:
For big blocks like this (which exceed my brain cache), perhaps I could suggest a loop?
```
for (auto name : { ID::A, ID::B, ID::C ... }) {
}
```

Reply:
That's a great idea :).